From d950fbb967d3692b31803d381059a64fb456b071 Mon Sep 17 00:00:00 2001
In-Reply-To: <20191115040909.GA21654@sigill.intra.peff.net>
References: <20191115040909.GA21654@sigill.intra.peff.net>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Thu, 14 Jan 2021 17:13:50 +0100
Subject: [PATCH 1/6] test-lib: add tests for test_might_fail
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This trivial sibling command of test_must_fail added in
fdf1bc48ca (t7006: guard cleanup with test_expect_success, 2010-04-14)
didn't have any tests. Let's add at least a basic one.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 t/t0000-basic.sh | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/t/t0000-basic.sh b/t/t0000-basic.sh
index f4ba2e8c85..f90c375fe9 100755
--- a/t/t0000-basic.sh
+++ b/t/t0000-basic.sh
@@ -1326,4 +1326,10 @@ test_expect_success 'test_must_fail rejects a non-git command with env' '
 	grep -F "test_must_fail: only '"'"'git'"'"' is allowed" err
 '
 
+test_expect_success 'test_might_fail is like test_must_fail ok=' '
+	! test_must_fail git version &&
+	! test_must_fail ok= git version &&
+	test_might_fail git version
+'
+
 test_done
-- 
2.29.2.222.g5d2a92d10f8

